src/packet_distribution.cpp

```diff
@@ -40,36 +40,33 @@ void AOApplication::server_packet_received(AOPacket packet)
 
   if (header == "decryptor")
   {
-    if (content.size() == 0)
+    if (content.size() < 1)
     {
       return;
     }
 
-    // default(legacy) values
-    m_serverdata.set_features(QStringList());
-
-    QString f_hdid;
-    f_hdid = get_hdid();
+    net_manager->server_connected(true);
 
-    QStringList f_contents = {f_hdid};
-    AOPacket hi_packet("HI", f_contents);
-    send_server_packet(hi_packet);
-    log_to_demo = false;
+    QStringList f_contents = {get_protocol_version_string(), "AO2", get_version_string()};
+    send_server_packet(AOPacket("ID", f_contents));
   }
   else if (header == "ID")
   {
-    if (content.size() < 2)
+    if (content.size() < 3)
```

Comment:
i think this causes the playercount to stop working

Reply:
Protocol changes, intended.